pkg/cloud/gcp/actuators/machine/actuator.go

```diff
@@ -96,7 +97,28 @@ func (a *Actuator) Exists(ctx context.Context, machine *machinev1.Machine) (bool
 	// When create()/update() try to store machineSpec/status this might result in
 	// "Operation cannot be fulfilled; the object has been modified; please apply your changes to the latest version and try again."
 	// Therefore we don't close the scope here and we only store spec/status atomically either in create()/update()"
-	return newReconciler(scope).exists()
+	invalidMachineConfiguration, exists, err := newReconciler(scope).exists()
+	if !invalidMachineConfiguration {
+		return exists, err
+	}
+
+	if exists {
+		machine.Status.Phase = pointer.String("Failed")
+	} else {
+		// If the machine has, e.g. invalid zone, and it doesn't have a phase set yet,
+		// we have to make sure the phase goes as "Failed".
+		if machine.Status.Phase == nil {
+			machine.Status.Phase = pointer.String("Failed")
+		}
```

Comment:
Returning an invalid machine configuration error doesn't set the phase to failed?